### PR TITLE
React.createElement splits SVG/HTML props based on type parameter

### PR DESCRIPTION
**Pull request by sandersn**
This is required for 2.4 to avoid a weak type error since DOMAttributes is a weak type.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#typescript-24
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react/index.d.ts [hunk 1]
```diff
@@ -136,8 +136,12 @@ declare namespace React {
         type: ClassType<P, T, C>): CFactory<P, T>;
     function createFactory<P>(type: ComponentClass<P>): Factory<P>;
 
-    function createElement<P extends DOMAttributes<T>, T extends Element>(
-        type: string,
+    function createElement<P extends HTMLAttributes<T>, T extends Element>(
+        type: keyof ReactHTML,
+        props?: ClassAttributes<T> & P,
+        ...children: ReactNode[]): DOMElement<P, T>;
```
Comment (vsiao) [inline types/react/index.d.ts]: this can return `ReactHTMLElement` now, correct?

File: types/react/index.d.ts [hunk 2]
```diff
@@ -136,8 +136,12 @@ declare namespace React {
...
+    function createElement<P extends SVGAttributes<T>, T extends Element>(
+        type: keyof ReactSVG,
         props?: ClassAttributes<T> & P,
         ...children: ReactNode[]): DOMElement<P, T>;
```
Comment (vsiao) [inline types/react/index.d.ts]: and `ReactSVGElement` here.

Comment (sandersn) [inline types/react/index.d.ts]: Yes

File: types/react/index.d.ts (hunk 1 above)
Comment (sandersn) [inline types/react/index.d.ts]: Yes, but only if `T` is restricted to extend `HTMLElement`. I think this is fine to change, though.

File: types/react/index.d.ts [hunk 3]
```diff
@@ -136,10 +136,14 @@ declare namespace React {
...
     function createFactory<P>(type: ComponentClass<P>): Factory<P>;
 
-    function createElement<P extends DOMAttributes<T>, T extends Element>(
-        type: string,
+    function createElement<P extends HTMLAttributes<T>, T extends HTMLElement>(
+        type: keyof ReactHTML,
         props?: ClassAttributes<T> & P,
-        ...children: ReactNode[]): DOMElement<P, T>;
+        ...children: ReactNode[]): ReactHTMLElement<T>;
+    function createElement<P extends SVGAttributes<T>, T extends Element>(
```
Comment (vsiao) [inline types/react/index.d.ts]: `T extends SVGElement`

**Review by vsiao — APPROVED** [review]
👍

**Review by Jessidhia — APPROVED** [review]

**Comment by johnnyreilly** [comment]
Thanks!

**Pull request merged**